src/main/java/randoop/main/GenTests.java

```diff
@@ -516,6 +522,75 @@ public boolean handle(String[] args) {
           codeWriter,
           GenInputsAbstract.regression_test_basename,
           "Regression");
+
+      HashSet<String> flakyTestNames = codeWriter.getFlakyTestNames();
+
+      if (flakyTestNames.size() > 0) {
+        List<ExecutableSequence> regressionSequences = explorer.getRegressionSequences();
+
+        // How many tests an operation occurs in.
+        // Tally occurrences of operations in all tests.
+        // Each method is counted exactly once if it appears in a sequence test.
+        Map<TypedOperation, Integer> testOccurrences =
+            tallyOperationsInSequences(regressionSequences);
+
+        // How many flaky tests an operation occurs in.
+        Map<TypedOperation, Integer> flakyOccurrences;
+
+        // TODO: cxing handle Error Test Sequence tallying.
+        // tallyOperationsInSequences(testOccurrences, explorer.getErrorTestSequences());
+
+        List<ExecutableSequence> flakySequences = new ArrayList<>();
+        for (String flakyTestName : flakyTestNames) {
+          int testNum = Integer.parseInt(flakyTestName.substring(4)); // length of "test"
+          // Tests start at 001
+          ExecutableSequence flakySequence = regressionSequences.get(testNum - 1);
+          flakySequences.add(flakySequence);
+        }
+
+        // Tally occurrences of operations in flaky tests.
+        // Each method is counted exactly once if it appears in a flaky test.
+        flakyOccurrences = tallyOperationsInSequences(flakySequences);
+
+        // tf-idf metric
+        // Our heuristic for probability of whether a method is flaky
+        //  is equal to the ratio of the number of flaky tests a method M occurs
+        //  in divided by the number of total tests M occurs in.
+        // The result is multiplied by 100.0 to obtain a percentage.
+
+        // Maps from a fully-qualified method signature to the above
+        //  percentage.
+        Map<String, Double> methodFlakyPercentage = new HashMap<>();
+
+        for (TypedOperation to : testOccurrences.keySet()) {
+          double flakyMethodOccurrences =
+              flakyOccurrences.containsKey(to) ? flakyOccurrences.get(to) : 0;
+          methodFlakyPercentage.put(
+              to.toParsableString(), flakyMethodOccurrences / testOccurrences.get(to));
+        }
+
+        // Sort in descending order by value.
+        List<Entry<String, Double>> sortedMethodsByFlakiness =
+            new ArrayList<>(methodFlakyPercentage.entrySet());
+        sortedMethodsByFlakiness.sort(reverseOrder(Entry.comparingByValue()));
+
+        System.out.println();
+        System.out.println("Flaky tests were generated. The following methods,");
```

Comment:
This speaks about a culprit; it would be helpful to give a simple explanation such as "This means that your program contains side-effecting or nondeterministic methods."
Rather than "Please determine", give guidance about what to do.  I think it is adequate to give a link to the manual, such as https://randoop.github.io/randoop/manual/index.html#nondeterminism .  Be sure the manual is adequate.